### DISCUSSION TITLE
Seek Forward and Backward

### DISCUSSION THREAD

**Discussion by Kyokuhin-b** [MiczFlor/RPi-Jukebox-RFID#1827] [Q&A]
Hi everybody,

i´m new on this Forum and I'm a beginner with Phoniebox.

I tried to implement a new button to seek Forward and Backward.
But on "RPi-Jukebox-RFID/settings/gpio_settings.ini" I Just can use "VolumenControl", "PrevNextContro"l and "PlayPause" on the standart PINs. The other Pins doesn't works.

So I think I have to activate the diferents PINs on "RPi-Jukebox-RFID/scripts/gpio-buttons.py", but I can't find the "gpio-buttons.py".

My next step is to replace two Buttons, one for NextTrack and one for SeekFwd with one which do both things. NextTrack with a simple press and SeekFwd holding the Butoons.

Can somobody help me or maybe it is implemented yet but i don't know about this feature?

Thank you for your helps!!!

**Comment by s-martin** [comment]
Please have a look at the examples folder:
https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/develop/components/gpio_control/example_configs

gpio_buttons.py doesn't exist anymore, it's all configured via ./settings/gpio_settings.ini


The example file should help you to find configurations for your need. Basically you can use any GPIO pin.

**Comment by Kyokuhin-b** [comment]
Hi and thank you!

I know the Example Folder but the pins also don’t working for me

**Comment by s-martin** [comment]
I guess you have already seen https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Using-GPIO-hardware-buttons

One important thing to note: for `Pin` you need to enter the **GPIO number**!

**Comment by Kyokuhin-b** [comment]
Hi,

I found my problem. The buttons wasn't enabled...

But I still habe problems. I can't change the hold-time from 0.3 to another value like 2

**Comment by s-martin** [comment]
What are you trying to do exactly and what doesn't work?

**Comment by Kyokuhin-b** [comment]
So, I'll use the translator now, I hope that you will understand me better that way and that I am not sitting on a message for what feels like 3 hours.

I have integrated a button each for "PlayerNext" and "PlayerPrev".

Now I would like to realize that if you press it briefly, you fast-forward ("seekFwd") and if you hold it down, it jumps to the next track ("PlayerNext"). Unfortunately, I can't get the PIN assigned twice. I guess I have to program it, unfortunately I don't really know which file is in which it has to be done.

Unfortunately I haven't found this project anywhere.

The other thing I meant was that I tried in the file:
"RPi-Jukebox-RFID / settings / gpio_settings.ini"
to extend the duration of the key press until the function is executed.
So I increased the "hold_time" value to "1". Nevertheless, he always switched after a few milliseconds.

Can you please help me with this?

**Reply by AlvinSchiller** [reply]
This should be with possible the `hold_mode` `SecondFunc` of a button.

There now is also a fix with https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2367 in v2.8.0.
The behavior of `hold_mode` `SecondFunc` and `SecondFuncRepeat` has changed. The secondary function is no longer triggered additionally to the primary function.
Now its called exclusively if `hold_time` is reached. The primary function will only be triggered if the button is pressed shorter then `hold_time`! Existing configurations may need to adapt to this.

**Comment by splitti** [comment]
I am also interested in a solution for different function-calls for short and long pressed buttons. This was possible in the old version...  And also for function calls for two pressed buttons.
